include/onnxruntime/core/framework/op_kernel.h

```diff
@@ -129,6 +131,37 @@ class OpKernel {
     return Status::OK();
   }
 
+  // Override this function to get pre-packed tensors from this kernel.
+  // Only useful for models run on PC with CPU so ORT could load prepacked weights directly from
+  // ONNX data file with mmap and no need to do prepacking on fly to save a lot of heap memory.
+  // For example, matmul_nbits kernel can use below function to get prepacked weights as tensor.
+  // template <typename T1>
+  // std::unique_ptr<Tensor> MatMulNBits<T1>::GetPrePackTensors() {
+  //   return std::move(packed_tensor_);
+  // }
+  // Please refer to matmul_nbits kernel for a complete example
+  virtual std::unique_ptr<Tensor> GetPrePackTensors() {
```

Comment:
Tensor is a very shallow, movable class. In a way, it is a smart ptr itself. There is not a reason to create it on a heap and wrap it in another smart ptr.

It is exposed in another providers via smart ptr indirection, but there is not a reason to do so in the CPU code.

Reply:
I see, just want to wrap Tensor with smart pointer to avoid memory leak, let me think about it.

Reply:
Well, if you deal with objects by value, there is not a way for them to leak.